jenkins/image_building/build-ci-image.sh

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+
+set -eux
+
+export IMAGE_OS="${IMAGE_OS}"
+
+current_dir="$(dirname "$(readlink -f "${0}")")"
+
+# Disable needrestart interactive mode
+sudo sed -i "s/^#\$nrconf{restart} = 'i';/\$nrconf{restart} = 'a';/" /etc/needrestart/needrestart.conf   > /dev/null
+
+sudo apt-get update
+
+# Install packages
+sudo apt-get install python3-pip qemu qemu-kvm -y
+sudo pip3 install diskimage-builder python-openstackclient
+
+export ELEMENTS_PATH="${current_dir}/dib_elements"
+export DIB_DEV_USER_USERNAME="metal3ci"
+export DIB_DEV_USER_PWDLESS_SUDO="yes"
+export DIB_DEV_USER_AUTHORIZED_KEYS="${current_dir}/id_ed25519_metal3ci.pub"
+
+if [[ "${IMAGE_OS}" == "ubuntu" ]]; then
+  export DIB_RELEASE=jammy
+else
+  export DIB_RELEASE=9
+fi 
+
+# Set image names
+commit_short="$(git rev-parse --short HEAD)"
+img_date="$(date --utc +"%Y%m%dT%H%MZ")"
+
+final_ci_img_name="metal3-ci-${IMAGE_OS}"
+ci_img_name="${final_ci_img_name}-${img_date}-${commit_short}"
+
+# Create an image
+disk-image-create --no-tmpfs -a amd64 "${IMAGE_OS}"-ci "${IMAGE_OS}" -o "${ci_img_name}" block-device-efi
+
+# Push image to openstack
+openstack image create "${final_ci_img_name}" --file "${ci_img_name}".qcow2 --disk-format=qcow2
```

Comment:
Shouldn't we use `ci_img_name` here instead?
```suggestion
openstack image create "${ci_img_name}" --file "${ci_img_name}".qcow2 --disk-format=qcow2
```

Reply:
`final_ci_image_name`: `metal3-ci-ubuntu`
ci_image_name`: `metal3-ci-ubuntu-20240220T2103Z-1339d8d`

My idea was here to have one image name `metal3-ci-ubuntu` which is latest.
And for backup I wanted to use `metal3-ci-ubuntu-20240220T2103Z-1339d8d` this naming style. 

We will do back up images as a next step.

Reply:
Ok I see, we directly overwrite the latest and can then optionally revert to a backup. That is fine. Actually now that I think about it the name makes more sense in this way. I was wondering why it was "final" when we then used `ci_image_name` :joy: